### PR TITLE
Wito/feature/transfer single resource

### PR DESCRIPTION
**Pull request by WitoDelnat**
This PR fixes some bugs and adds single resource transfer from either overview or inspect view.

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: src/redux/services/compare/transferResource.ts [hunk 1]
```diff
@@ -101,21 +101,18 @@ async function deployResourceToCluster(
...
 ): Promise<K8sResource> {
-  const resourceMap = state.main.resourceMap;
-  const fileMap = state.main.fileMap;
-
   if (target) {
-    const result = {...target};
-    performResourceContentUpdate(result, source.content, fileMap, resourceMap);
+    const result = structuredClone(target);
+    result.text = source.text;
+    await dispatch(updateResource({resourceId: target.id, text: source.text}));
```
Comment (WitoDelnat) [inline src/redux/services/compare/transferResource.ts:109]: This should already fix the bug where I mutate frozen state. A bulk approach is still in the back of my mind but with lower priority.

Comment (devcatalin) [reply src/redux/services/compare/transferResource.ts:109]: I didn't know you can `await` dispatches, interesting!

**Review by devcatalin — APPROVED** [review]

File: src/redux/thunks/updateResource.ts [hunk 2]
```diff
@@ -2,31 +2,35 @@ import {createAsyncThunk, createNextState, original} from '@reduxjs/toolkit';
...
 import {isKustomizationPatch, isKustomizationResource, processKustomizations} from '@redux/services/kustomize';
 import {getK8sVersion} from '@redux/services/projectConfig';
 import {reprocessResources} from '@redux/services/resource';
 import {findResourcesToReprocess} from '@redux/services/resourceRefs';
 import {updateSelectionAndHighlights} from '@redux/services/selection';
 
-export const updateResource = createAsyncThunk(
+type UpdateResourcePayload = {
+  resourceId: string;
+  text: string;
```
Comment (topliceanurazvan) [inline src/redux/thunks/updateResource.ts:19]: Great catch here for renaming content to text ! 🙌🏻

**Review by topliceanurazvan — APPROVED** [review]
LGTM! Great work

**Pull request merged**